src/main/scala/gitbucket/core/service/RepositoryService.scala

```diff
@@ -434,7 +434,7 @@ trait RepositoryService { self: AccountService =>
     val preferred = PluginRegistry().renderableExtensions.map(extension => s"${fileBaseName}.${extension}")
 
     def choiceTemplate(files: List[FileInfo]): Option[FileInfo] =
-      files.find(f => preferred.contains(f.name)).orElse(files.find(f => f.name.startsWith(fileBaseName)))
+      files.find(f => preferred.contains(f.name)).orElse(files.find(f => fileBaseName == FileUtil.trimExtension(f.name)))
```

Comment:
sorry by just commenting the diffs I missed the real type.

I just tested how it really behaves on github in [McFoggy/jira-example-project](https://github.com/McFoggy/jira-example-project).

According to my few tests github trys to load files in the following order:
- FILE
- FILE.[whatever extension] where extensions are order by lexicographic order

Thus regarding your PR I would just order the preferred list to pickup the first defined renderer, and then try to find a file which name starts with fileBaseName.`  (notice the dot)